### PR TITLE
Filtering and paging of Join Forms

### LINKED ISSUE #1125: Join submission pagination and filtering
Implement pagination and filtering of join submissions (based on form and state).

Related to #1101.

### PR DESCRIPTION
**Pull request by niklasva82**
Resolves #1125

### CODE REVIEW

**Comment by richardolsson** [comment]
Note to self while reviewing (and perhaps to you @niklasva82 in case you didn't think of this already).

Deleting a person should update the store and remove all join submissions matching that person, and ideally close any related JoinSubmissionPanes. Steps to test:

1. Open a join submission pane from the list
2. Approve if not approved already
3. Click the person to open a person pane
4. Edit person date
5. Delete person

Expected results: The join submission should disappear from the list, and all three panes should close.

**Review by richardolsson — CHANGES_REQUESTED** [review]
Code looks good and seems to work as expected. The only thing is that I'd like the filter button to move. Does that make sense?

image

**Comment by niklasva82** [comment]
> Code looks good and seems to work as expected. The only thing is that I'd like the filter button to move. Does that make sense?
> 
> image

Sure! That actually 

> Code looks good and seems to work as expected. The only thing is that I'd like the filter button to move. Does that make sense?
> 
> image

Done, that also removed the need for changes in RootPaneBase :)

**Comment by richardolsson** [comment]
> Done, that also removed the need for changes in RootPaneBase :)

Ah, I see! Maybe remove the CSS also then?

File: src/components/sections/people/JoiningPane.scss [hunk 1]
```diff
@@ -11,4 +11,35 @@
...
+    .RootPaneBase-toolbar-column-reverse {
+        display: flex;
+        flex-direction: column-reverse;
+        margin-left: -1em;
+
+        .RootPaneBase-filterButton {
+            margin-top: 0.5em;
+            margin-left: 0.8em;
+        }
+    }
```
Comment (richardolsson) [inline src/components/sections/people/JoiningPane.scss:44]: Is this still needed? Or only part of it?

**Review by richardolsson — CHANGES_REQUESTED** [review]

File: src/components/sections/people/JoiningPane.scss (hunk 1 above)
Comment (niklasva82) [reply src/components/sections/people/JoiningPane.scss]: Yeah the toolbar-column-reverse is no longer needed. Fixed.

**Review by richardolsson — APPROVED** [review]
Nice! We've arrived!

**Pull request merged**